ipa-core/src/query/runner/hybrid.rs

```diff
@@ -175,6 +180,36 @@ where
     }
 }
 
+pub async fn execute_hybrid_protocol<'a, R: PrivateKeyRegistry>(
+    prss: &'a Endpoint,
+    gateway: &'a Gateway,
+    input: BodyStream,
+    ipa_config: HybridQueryParams,
+    config: &QueryConfig,
+    key_registry: Arc<R>,
+) -> QueryResult {
+    let gate = Gate::default();
+    let cross_shard_prss =
+        setup_cross_shard_prss(gateway, &gate, prss.indexed(&gate), gateway).await?;
+    let sharded = Sharded {
+        shard_id: gateway.shard_id(),
+        shard_count: gateway.shard_count(),
+        prss: Arc::new(cross_shard_prss),
+    };
+
+    let ctx = ShardedMaliciousContext::new_with_gate(prss, gateway, gate, sharded);
+
+    // TODO: load this from individual reports
+    let hybrid_info =
```

Comment:
@tyurek is it something you would need to fix along with the AAD change?

Reply:
actually this should be fixed. I'll find time to rebase this evening